src/bios_handler.cpp

```diff
@@ -362,8 +365,91 @@ void IbmBiosHandler::processCreateDefaultLpar()
         saveCreateDefaultLparToBios(*l_pVal);
         return;
     }
-
     logging::logMessage(
         "Invalid type recieved for create default Lpar from VPD.");
 }
+
+void IbmBiosHandler::saveClearNvramToVpd(const std::string& i_clearNvramVal)
+{
+    if (i_clearNvramVal.empty())
+    {
+        logging::logMessage(
+            "Empty value received for clear NVRAM from BIOS. Skip updating to VPD.");
+        return;
+    }
+
+    // Read required keyword from DBus as we need to set only a Bit.
+    auto l_kwdValueVariant = dbusUtility::readDbusProperty(
+        constants::pimServiceName, constants::systemVpdInvPath,
+        constants::utilInf, constants::kwdClearNVRAM_CreateLPAR);
+
+    if (auto l_pVal = std::get_if<std::string>(&l_kwdValueVariant))
+    {
+        types::BinaryVector l_valToUpdateInVpd;
+        if (i_clearNvramVal.compare("Enabled") == constants::STR_CMP_SUCCESS)
```

Comment:
I would suggest against this compare style , would be better if you can make the inputs string to  lower case and compare to "enabled", so that if they change the case later on we will not be affected by it.

Reply:
maybe need to do this for the Lpar one also